### PR TITLE
Localstorage

### PR DESCRIPTION
**Pull request by RickKollenbroich93**
Implement localstorage hook

- Create useCollection custom hook with localstorage hook
- Create useAddCard custom hook with localstorage hook
- Change CodeCard and CodeField component for better usage
- Change types and create an types.ts for typing Collections and CodeCards
- Uplift many states from lowest component up to the page where the component is used in
- Remove some dead components which isn´t needed any more

### CODE REVIEW

File: src/app/components/CodeField/CodeField.tsx [hunk 1]
```diff
@@ -8,18 +7,22 @@ import { ScrollSync, ScrollSyncPane } from 'react-scroll-sync';
...
 };
 
 export default function CodeField({
   layout = 'compact',
   language,
+  content,
+  onChange,
 }: CodeFieldProps): JSX.Element {
-  const [content, setContent] = useState('//Your Highlighted Code');
+  // const [content, setContent] = useState('//Your Highlighted Code');
```
Comment (phlgr) [inline src/app/components/CodeField/CodeField.tsx]:
```suggestion
```

File: src/app/components/CodeField/CodeField.tsx [hunk 2]
```diff
@@ -8,18 +7,22 @@ import { ScrollSync, ScrollSyncPane } from 'react-scroll-sync';
...
 
-  function handleChange(event: ChangeEvent<HTMLTextAreaElement>) {
-    const { value } = event.target;
-    setContent(value);
-  }
+  // function handleChange(event: ChangeEvent<HTMLTextAreaElement>) {
+  // //    const { value } = event.target;
+  // //    setContent(value);
+  // //  }
```
Comment (phlgr) [inline src/app/components/CodeField/CodeField.tsx]:
```suggestion
```

File: src/app/pages/AddCard/AddCard.tsx [hunk 3]
```diff
@@ -1,11 +1,66 @@
...
+      language: collection.language,
+      onClick: () => addToCard(collection),
+    };
+  });
+  function addToCard(collection: Collection) {
+    const addedCollection = {
+      children: collection.name,
+      language: collection.language,
+    };
+    console.log(addedCollection);
```
Comment (phlgr) [inline src/app/pages/AddCard/AddCard.tsx]:
```suggestion
```

File: src/app/pages/AddCard/AddCard.tsx [hunk 4]
```diff
@@ -1,11 +1,66 @@
...
+    const allSelectedCollections = [...selectedCollections, addedCollection];
+    setSelectedCollections(allSelectedCollections);
+  }
+
+  const { addCodeCard } = useCodeCard();
```
Comment (phlgr) [inline src/app/pages/AddCard/AddCard.tsx]: hooks should always be called at the top

File: src/app/pages/AddCard/AddCard.tsx [hunk 5]
```diff
@@ -15,12 +70,26 @@ export default function AddCard(): JSX.Element {
...
+        <CodeField
+          language={selectedLanguage}
+          content={content}
+          onChange={handleChange}
+        />
+        <CollectionTagGroup
+          allCollections={tagCollections}
+          selectedCollections={selectedCollections}
+        />
+        <AddButton type="submit" children="Add new Card" onClick={addCard} />
```
Comment (phlgr) [inline src/app/pages/AddCard/AddCard.tsx]:
```suggestion
        <AddButton type="submit" children="Add new Card" />
```

File: src/app/pages/AddCard/AddCard.tsx [hunk 6]
```diff
@@ -15,12 +70,26 @@ export default function AddCard(): JSX.Element {
           NewCard
         </Header>
       </section>
-      <CodeForm
-        className={styles.codeForm}
-        handleSubmit={() => console.log('Submitted')}
-        children="Add new Card"
-        onChange={(event) => console.log(event.target.value)}
-      />
+      <form className={styles.formWrapper}>
```
Comment (phlgr) [inline src/app/pages/AddCard/AddCard.tsx]:
```suggestion
      <form className={styles.formWrapper} handleSubmit={event=>{
        event.prevenDefault();
        addCard()
      }}>
```

**Review by phlgr — CHANGES_REQUESTED** [review]

File: src/app/components/CodeField/CodeField.tsx (hunk 1 above)
Comment (RickKollenbroich93) [reply src/app/components/CodeField/CodeField.tsx]: this state is uplifted so i terminate the whole line

**Pull request merged**